### PR TITLE
Fix symbol numbering bug

### PR DESCRIPTION
**Pull request by lukaszcz**
This should fix the bug with disappearing definitions.

### CODE REVIEW

File: src/Juvix/Compiler/Core/Data/InfoTable.hs [hunk 1]
```diff
@@ -12,7 +12,9 @@ data InfoTable = InfoTable
     _infoIdentifiers :: HashMap Symbol IdentifierInfo,
     _infoInductives :: HashMap Symbol InductiveInfo,
     _infoConstructors :: HashMap Tag ConstructorInfo,
-    _infoAxioms :: HashMap Name AxiomInfo
+    _infoAxioms :: HashMap Name AxiomInfo,
+    _infoSymbolsNum :: Word,
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Core/Data/InfoTable.hs:16]: I would prefer the name `infoNextSymbol` instead or something similar. It clarifies that this field is only used for fresh names and does never need to be decreased.
Same for tags.

File: src/Juvix/Compiler/Core/Data/InfoTableBuilder.hs [hunk 2]
```diff
@@ -26,52 +26,35 @@ checkSymbolDefined sym = do
...
-        modify' (over stateInfoTable (over identMap (HashMap.insert (ci ^. (constructorName . nameText)) (IdentTag (ci ^. constructorTag)))))
+        modify' (over infoConstructors (HashMap.insert (ci ^. constructorTag) ci))
+        modify' (over identMap (HashMap.insert (ci ^. (constructorName . nameText)) (IdentTag (ci ^. constructorTag))))
       RegisterIdentNode sym node ->
-        modify' (over stateInfoTable (over identContext (HashMap.insert sym node)))
+        modify' (over identContext (HashMap.insert sym node))
       SetIdentArgsInfo sym argsInfo -> do
-        modify' (over stateInfoTable (over infoIdentifiers (HashMap.adjust (set identifierArgsInfo argsInfo) sym)))
-        modify' (over stateInfoTable (over infoIdentifiers (HashMap.adjust (set identifierArgsNum (length argsInfo)) sym)))
+        modify' (over infoIdentifiers (HashMap.adjust (set identifierArgsInfo argsInfo) sym))
```
Comment (janmasrovira) [inline src/Juvix/Compiler/Core/Data/InfoTableBuilder.hs:54]: with lenses:
```
        modify' (set (infoIdentifiers . at sym . _Just . identifierArgsInfo) argsInfo)
        modify' (set (infoIdentifiers . at sym . _Just . identifierArgsNum) (length argsInfo))
```

File: src/Juvix/Compiler/Core/Data/InfoTable.hs (hunk 1 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Core/Data/InfoTable.hs:16]: Done

File: src/Juvix/Compiler/Core/Data/InfoTableBuilder.hs (hunk 2 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Core/Data/InfoTableBuilder.hs:54]: Okay.

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**